src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -102,7 +102,10 @@ void CoreWorkerDirectTaskReceiver::HandleTask(
         ObjectID id = ObjectID::FromIndex(task_spec.TaskId(), /*index=*/i + 1);
         return_object->set_object_id(id.Binary());
 
-        // The object is nullptr if it already existed in the object store.
+        // This should only happen if the local raylet died. Caller should
+        // retry the task.
+        RAY_CHECK(return_objects[i])
```

Comment:
Hmm does this mean if the local raylet is dead, the worker can just crash here? 

Can we handle this more gracefully without the error message?

Reply:
I did this on purpose so that the caller can handle the error.

Reply:
Oh I meant this will generate the ugly cpp stacktrace when this happens to the driver (agreed caller should handle the error). 

Maybe we can just do RAY_LOG(WARNING) and QuickExit() instead?

Reply:
Let's do this as part of a larger refactor. There are a lot of other RAY_CHECKs in this file and there's no way to access QuickExit() from here.

Reply:
Actually QuickExit is inside util.h. Is it still difficult in this case? I am okay with doing this in other PRs

Reply:
Ah I see, my mistake! Thanks yeah I'll do it here.